sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala

```diff
@@ -1095,13 +1105,11 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
      * relationship. Each ColumnarBatch is converted to a single ParquetCachedBatch when next()
      * is called on this iterator
      */
-    class ColumnarBatchToCachedBatchIterator extends InternalRowToCachedBatchIterator {
-      override def getIterator: Iterator[InternalRow] = {
-
-        new Iterator[InternalRow] {
-          // We have to check for null context because of the unit test
-          Option(TaskContext.get).foreach(_.addTaskCompletionListener[Unit](_ => hostBatch.close()))
+    class ColumnarBatchToCachedBatchIterator extends InternalRowToCachedBatchIterator
+        with AutoCloseable {
+      val hostBatches = new ListBuffer[ColumnarBatch]()
```

Comment:
Nothing is ever added to `hostBatches`.

Reply:
Thanks, I have added a test to make sure we throw